File: algo-c-to-_/src/lissajouscurve.py
#
#	from src/lissaj.c
#
#	a part of main		to	lissajousCurve
#

from math import pi, sin, cos

def lissajousCurve(plotter, n, offset):
    stepX = lambda m: n + offset + n * cos(m)
    stepY = lambda m: n + offset + n * sin(m)

    plotter.move(stepX(0), stepY(0))
    for i in range(1, 361):
        plotter.draw(stepX(3 * (pi / 180) * i), stepY(5 * (pi / 180) * i))

File: porting-from-algo-c/examples/lissajouscurve.lua
--
--  from src/lissaj.c
--
--    a part of main  to  lissajousCurve
--

local M = require 'svgplot'

local SvgPlotA = M.SvgPlot
local SvgPlotB = M.SvgPlotWholeBuffer
local SvgPlotC = M.SvgPlotWithBuffer
local lissajousCurve = require 'lissajouscurve'.lissajousCurve
local file = require '_helper'.file

do
	local n, offset = 300, 10

	function body(plotter)
		plotter:pathStart()
		lissajousCurve(plotter, n, offset)
		plotter:pathEnd(false, M.SV.PRESET_PLAIN)
	end

	local m = (n + offset) * 2

	file("results/lissajouscurve-A.svg", "w", function (fh)
		SvgPlotA(m, m):write(fh, body)
	end)

	file("results/lissajouscurve-B.svg", "w", function (fh)
		SvgPlotB(m, m):write(fh, body):reset()
	end)

	file("results/lissajouscurve-C.svg", "w", function (fh)
		SvgPlotC(m, m):write(fh, body, 30)
	end)
end
